api/dsap/relationships.js

```diff
@@ -1,22 +1,67 @@
 'use strict';
 module.exports = function (context) {
     var relationships = context.models.relationships;
+    var models = context.models;
 
-    return {
+    const functions = {
+        mapArtistToSubject: function (aritistId) {
+            // TODO Map artist to subject.
+            return 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11';
+        },
+        mapWorkToSubject: function (workId) {
+            // TODO Map work to subject.
+            return 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11';
+        },
         findAllRelationships: function () {
             return relationships.findAll({
                 attributes: [
                     'id'
                 ]
             });
         },
+        findAllRelationshipsBySubject: function (subject, relation, object, offset = 0, limit = 20) {
+            var subjectWhere = subject ? { id: subject } : undefined;
+            var objectWhere = object ? { id: object } : undefined;
+            var relationWhere = relation ? relation : undefined;
+
+            return relationships.findAll({
+                attributes: [
+                    'id',
+                    'confidence',
+                    'relation'
+                ],
+                where: {
+                    relation: relationWhere
+                },
+                include: [
+                    { model: models.relationshipDescriptions },
+                    { model: models.relationshipOccurrences },
+                    { model: models.relationshipEntities, as: 'Subject', where: subjectWhere},
+                    { model: models.relationshipEntities, as: 'Object', where: objectWhere }
+                ],
+                offset,
+                limit
+            });
+        },
+        findSuggestions: function (subject, query, offset = 0, limit = 20) {
+            // TODO Perform an actual search (Using Elasticsearch? PostgreSQL?)
```

Comment:
@Henni Maybe for starters, you can use [pg_trgm](https://www.postgresql.org/docs/current/static/pgtrgm.html) to get suggestions for words. Just `sequelize.fn` the `similarity`, and get the top 5…

I'm not really sure, how well `ts_query` can map to sequelize…